File: apps/mobile/lib/screens/admin/admin_launch_overview.dart
import 'package:flutter/material.dart';

/// Read-only summaries. Actions delegate to existing authorized Admin screens.
class AdminLaunchOverview extends StatelessWidget {
  const AdminLaunchOverview({
    required this.data,
    this.onBilling,
    this.onProviders,
    this.onAgents,
    this.onSocial,
    super.key,
  });
  final Map<String, dynamic> data;
  final VoidCallback? onBilling, onProviders;
  final VoidCallback? onAgents, onSocial;
  Map<String, dynamic> map(Object? v) =>
      v is Map ? Map<String, dynamic>.from(v) : {};
  String count(Object? v) => v is num ? '$v' : 'Unavailable';
  String state(Object? v) => switch (v) {
    'completed' || 'complete' || 'healthy' || 'connected' => 'Healthy',
    'active' => 'Authorized',
    'available' => 'Saved status available',
    'paused' => 'Paused',
    'stale_lease' => 'Needs attention: previous run has not cleared',
    'connected_write' => 'Connected for publishing',
    'connected_read' => 'Connected for analytics',
    'not_authorized' => 'Automatic publishing not authorized',
    'running' || 'preparing' => 'Running',
    'held' => 'Held',
    'failed' || 'needs_attention' => 'Needs attention',
    'not_run' => 'No completed run recorded',
    _ => 'Unavailable',
  };
  String date(Object? v) => v is num
      ? DateTime.fromMillisecondsSinceEpoch(
          v.toInt(),
        ).toLocal().toString().split('.').first
      : 'Unavailable';
  Widget section(
    String title,
    List<Widget> children, {
    VoidCallback? action,
    String? actionLabel,
  }) => Card(
    child: Padding(
      padding: const EdgeInsets.all(16),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(
            title,
            style: const TextStyle(fontSize: 19, fontWeight: FontWeight.w600),
          ),
          const SizedBox(height: 8),
          ...children,
          if (action != null)
            TextButton(
              onPressed: action,
              child: Text(actionLabel ?? 'Open details'),
            ),
        ],
      ),
    ),
  );
  @override
  Widget build(BuildContext context) {
    final billing = map(data['billing']), email = map(data['email']);
    final runtime = map(data['runtime']), release = map(data['release']);
    final businesses = data['businesses'] is List
        ? data['businesses'] as List
        : const [];
    return Column(
      crossAxisAlignment: CrossAxisAlignment.stretch,
      children: [
        section('Marketplace readiness', [
          Text('New paid work: ${state(data['paidWork'])}'),
          if (data['payoutCertification'] is Map) ...[
            const Text(
              'LIVE Connect setup, genuine Scaler onboarding and payout readiness: Founder certified.',
            ),
            const Text('LIVE earning → cash-out → bank receipt: pending.'),
            const Text(
              'Certification history is not a live provider-readiness override.',
            ),
          ] else
            const Text('Payout certification evidence unavailable.'),
        ]),
        section(
          'Billing and accounts',
          [
            Text('Active memberships: ${count(billing['active'])}'),
            Text('Complimentary memberships: ${count(billing['comped'])}'),
            Text('Payment issues: ${count(billing['paymentIssues'])}'),
            Text('Team seat/access issues: ${count(billing['teamIssues'])}'),
            Text(
              'Cancellation scheduled: ${count(billing['cancellationScheduled'])}',
            ),
            Text(
              'Reconciliation issues: ${count(billing['reconciliationIssues'])}',
            ),
            const Text(
              'Customer payments, worker reserves and platform revenue are separate. No gross-payment revenue total is inferred.',
            ),
          ],
          action: onBilling,
          actionLabel: 'Review subscriptions',
        ),
        section(
          'ScaledCircle internal Growth',
          [
            Text('Research: ${state(map(data['internalGrowth'])['status'])}'),
            Text(
              'Last run: ${date(map(map(data['internalGrowth'])['lastRun'])['completedAt'])}',
            ),
            Text(
              'Result: ${state(map(map(data['internalGrowth'])['lastRun'])['status'])}',
            ),
            Text(
              'New: ${count(map(map(data['internalGrowth'])['lastRun'])['newProspectCount'])} · Duplicates: ${count(map(map(data['internalGrowth'])['lastRun'])['duplicatesExcludedCount'])} · Unavailable sources: ${count(map(map(data['internalGrowth'])['lastRun'])['unavailableSources'])}',
            ),
            Text(
              'Next eligible research: ${date(map(data['internalGrowth'])['nextResearchAfter'])}',
            ),
            Text(
              'Awaiting review: ${count(map(map(data['internalGrowth'])['summary'])['awaitingApproval'])}',
            ),
            const Text(
              'Separate internal workspace. Its results do not certify Attractive Remodel research.',
            ),
          ],
          action: onAgents,
          actionLabel: 'Open internal Growth workspace',
        ),
        section('Growth and Social workspaces', [
          if (businesses.isEmpty)
            const Text('No workspace run summaries available.'),
          for (final item in businesses)
            Builder(
              builder: (context) {
                final b = map(item),
                    r = map(b['research']),
                    s = map(b['social']);
                return ExpansionTile(
                  tilePadding: EdgeInsets.zero,
                  title: Text(b['name'] as String? ?? 'Business workspace'),
                  subtitle: Text(
                    'Research: ${state(r['status'])} • Social: ${state(s['authorization'])}',
                  ),
                  children: [
                    ListTile(
                      title: Text('Last research: ${date(r['lastRunAt'])}'),
                      subtitle: Text(
                        'Next eligible run: ${date(r['nextRunAt'])}\nNew opportunities: ${count(r['newOpportunities'])} • Duplicates suppressed: ${count(r['duplicatesSuppressed'])}\nSource checks: ${count(r['sourceChecks'])} • Unavailable sources: ${count(r['unavailableSources'])}',
                      ),
                    ),
                    const Text(
                      'A completed cycle with zero new opportunities is a valid result.',
                    ),
                    ListTile(
                      title: Text('Social worker: ${state(s['workerStatus'])}'),
                      subtitle: Text(
                        'Last completed: ${date(s['lastRunAt'])}\nNext worker invocation: ${s['nextWorkerRun'] ?? 'Unavailable'}\nScheduled: ${count(s['scheduled'])} • Publishing: ${count(s['publishing'])}\nPublished: ${count(s['published'])} • Needs attention: ${count(s['needsAttention'])}',
                      ),
                    ),
                    Text(
                      'Scheduled under current strategy: ${count(s['managedScheduled'])}',
                    ),
                    const Text(
                      'Counts include historical/manual posts; they do not certify autonomous publication.',
                    ),
                    for (final connection
                        in (s['connections'] as List? ?? const []))
                      Text(
                        '${map(connection)['provider'] == 'instagram' ? 'Instagram' : 'Facebook'}: ${state(map(connection)['status'])}',
                      ),
                    ExpansionTile(
                      title: const Text('Technical evidence'),
                      children: [
                        SelectableText(
                          'Workspace: ${b['businessId']}\nResearch run: ${r['runId'] ?? 'Unavailable'}',
                        ),
                      ],
                    ),
                  ],
                );
              },
            ),
          const Text(
            'ScaledCircle internal research uses its separate internal workspace. Review it in Growth operations; it is not inferred from Attractive Remodel.',
          ),
          if (onAgents != null)
            TextButton(
              onPressed: onAgents,
              child: const Text('Open Growth operations'),
            ),
          if (onSocial != null)
            TextButton(
              onPressed: onSocial,
              child: const Text('Open Social operations'),
            ),
        ]),
        section(
          'Email and provider readiness',
          [
            Text(
              email['branding'] == 'verified_checkpoint'
                  ? 'Google branding: verified at the maintained launch checkpoint.'
                  : 'Google branding: unavailable.',
            ),
            const Text(
              'Gmail restricted-scope review remains pending. Founder demo video and external review are required before general customer onboarding.',
            ),
            if (email['mailboxes'] is List)
              for (final mailbox in email['mailboxes'] as List)
                Text(
                  'Saved mailbox connection: ${state(map(mailbox)['health'])}',
                ),
            const Text(
              'Saved connection status is not a fresh provider probe.',
            ),
            Text(
              'Notification delivery issues: ${count(data['notificationFailures'])}',
            ),
          ],
          action: onProviders,
          actionLabel: 'Review provider configuration',
        ),
        section('Release evidence', [
          Text('Control-plane readback: ${date(runtime['checkedAt'])}'),
          Text('Hosting: ${runtime['hostingVersion'] ?? 'Unavailable'}'),
          Text(
            'Hosting source: ${runtime['hostingSource'] ?? 'Unavailable in release metadata'}',
          ),
          Text('Rules: ${runtime['rulesVersion'] ?? 'Unavailable'}'),
          Text(
            'Function inventory: ${runtime['functions'] is List ? (runtime['functions'] as List).length : 'Unavailable'}',
          ),
          Text('Native checkpoint: ${release['checkedAt'] ?? 'Unavailable'}'),
          Text(
            'iOS: ${map(release['ios'])['version'] ?? 'Unavailable'} (${map(release['ios'])['build'] ?? '-'}) · ${map(release['ios'])['state'] ?? 'Unavailable'}',
          ),
          Text(
            'Android: ${map(release['android'])['version'] ?? 'Unavailable'} (${map(release['android'])['build'] ?? '-'}) · ${map(release['android'])['state'] ?? 'Unavailable'}',
          ),
          Text(release['note'] as String? ?? 'Native evidence unavailable.'),
          for (final blocker in (release['blockers'] as List? ?? const []))
            Text('Remaining: $blocker'),
          ExpansionTile(
            title: const Text('Function revisions and scheduled workers'),
            children: [
              for (final f in (runtime['functions'] as List? ?? const []))
                Text(
                  '${map(f)['name']}: ${map(f)['state']} · ${map(f)['revision'] ?? 'Unavailable'}',
                ),
              for (final w in (runtime['workers'] as List? ?? const []))
                Text(
                  '${map(w)['name']}: ${map(w)['state']} · Next: ${map(w)['nextRunAt'] ?? 'Unavailable'}',
                ),
            ],
          ),
        ]),
      ],
    );
  }
}

File: functions-admin-ops/admin_launch_overview.js
"use strict";

// Read-only projections for the existing Admin overview. Never copy credentials,
// message content, bank details or arbitrary provider responses into this view.
const LIMIT = 101;
const number = v => Number.isFinite(v) ? v : null;
const time = v => typeof v?.toMillis === 'function' ? v.toMillis() : number(v);
const label = v => typeof v === 'string' ? v.slice(0, 120) : null;
const rows = s => s.docs.map(d => ({...d.data(), docId:d.id}));
function count(items, predicate) { return items === null ? null : items.filter(predicate).length; }
function research(schedule, run, now) {
  if (!schedule) return {status:'unavailable',lastRunAt:null,nextRunAt:null};
  const lease=time(schedule.leaseUntil);
  return {status:schedule.enabled===false?'paused':lease>0&&lease<now?'stale_lease':
    schedule.lastStatus==='completed'?'completed':label(schedule.lastStatus)||'not_run',
    lastRunAt:time(schedule.lastCompletedAt),nextRunAt:time(schedule.nextRunAt),
    newOpportunities:number(run?.newProspectCount),duplicatesSuppressed:number(run?.duplicatesExcludedCount),
    sourceChecks:number(run?.sourceChecks),unavailableSources:number(run?.unavailableSources),
    runId:label(schedule.lastRunId)};
}
async function load({db,now=Date.now(),paidWorkEnabled=null,project=null}) {
  const unavailable=[];
  async function read(name, query=db.collection(name)) {
    try {const s=await query.limit(LIMIT).get();if(s.docs.length>=LIMIT){unavailable.push(name);return null;}return rows(s);}
    catch {unavailable.push(name);return null;}
  }
  async function doc(path) {try {const s=await db.doc(path).get();return s.exists?s.data():null;}catch {unavailable.push(path.split('/')[0]);return null;}}
  const [subscriptions,schedules,policies,cycles,jobs,mailboxes,workspaces,notifications,operations]=await Promise.all([
    'businessSubscriptions','customerResearchSchedules','socialManagedPolicies','socialManagedCycles',
    'socialGrowthJobs','businessMailboxes','businessWorkspaces','mobilePushReceipts','financialOperations',
  ].map(n=>read(n)));
  const teamInventories=workspaces===null?null:await Promise.all(workspaces.map(async w=>{
    if(!/^[A-Za-z0-9_-]{1,160}$/.test(w.docId))return null;
    const members=await read('businessWorkspaces/'+w.docId+'/members');
    const entitlement=subscriptions?.find(v=>v.docId===w.docId);
    if(!members||subscriptions===null)return null;
    const capacity=require('./business_workspace').seats(entitlement||{},now);
    const active=members.filter(m=>m.status==='active'&&m.uid!==w.docId);
    return {businessId:w.docId,seatsUsed:1+active.length,capacity,
      needsReview:active.some(m=>m.businessId!==w.docId||!Number.isInteger(m.seatIndex)||m.seatIndex<1||m.seatIndex>=capacity)||active.length+1>capacity};
  }));
  const ids=[...new Set([...(schedules||[]),...(policies||[]),...(cycles||[])].map(x=>x.businessUid||x.docId))];
  if(ids.length>20)unavailable.push("workspace_inventory_truncated");
  const businesses=await Promise.all(ids.slice(0,20).map(async uid=>{
    if(!/^[A-Za-z0-9_-]{1,160}$/.test(uid))return null;
    const [profile,user]=await Promise.all([doc('businessGrowthProfiles/'+uid),doc('users/'+uid)]);
    const schedule=schedules?.find(s=>s.docId===uid&&s.businessUid===uid);
    const run=schedule?.lastRunId&&/^[A-Za-z0-9_-]{1,160}$/.test(schedule.lastRunId)?await doc('agentRuns/'+schedule.lastRunId):null;
    const policy=policies?.find(p=>p.docId===uid&&p.businessUid===uid);
    const cycle=cycles?.find(c=>c.docId===uid&&c.businessUid===uid);
    const own=jobs===null?null:jobs.filter(j=>j.businessUid===uid);
    let managedScheduled=null;
    if(own&&policy){
      const pending=own.filter(j=>j.status==='scheduled');
      const approvalIds=[...new Set(pending.map(j=>j.approvalId))];
      if(approvalIds.length<=20){
        const approved=await Promise.all(approvalIds.map(id=>typeof id==='string'&&/^[A-Za-z0-9_-]{1,160}$/.test(id)?doc('socialGrowthApprovals/'+id):null));
        if(approved.every(Boolean))managedScheduled=pending.filter(j=>{const a=approved[approvalIds.indexOf(j.approvalId)];return a?.businessUid===uid&&a.managedPolicyId===policy.id&&a.authorizationSource==='approved_strategy';}).length;
      }
    }
    const connections=await Promise.all(['facebook','instagram'].map(async provider=>{
      const c=await doc('socialConnections/'+uid+'/providers/'+provider);
      return {provider,status:c?label(c.status):'unavailable',health:c?label(c.tokenHealth):null};
    }));
    return {businessId:uid,name:label(profile?.businessName||user?.companyName||user?.businessName)||'Business workspace',
      research:research(schedule,run?.businessUid===uid?run:null,now),
      social:{authorization:policy?label(policy.status):'not_authorized',lastRunAt:time(cycle?.finishedAt),
        workerStatus:cycle?.leaseUntil>0&&cycle.leaseUntil<now?'stale_lease':label(cycle?.status)||'unavailable',
        nextWorkerRun:null,connections,
        scheduled:count(own,j=>j.status==='scheduled'),publishing:count(own,j=>['publishing','processing'].includes(j.status)),
        published:count(own,j=>j.status==='published'),needsAttention:Array.isArray(cycle?.results)?cycle.results.filter(r=>r.status==='needs_attention').length:null,
        failedPublishingJobs:count(own,j=>['failed','blocked','needs_attention'].includes(j.status)),
        managedScheduled}};
  }));
  return {generatedAt:now,unavailableSources:[...new Set(unavailable)],boundedInventoryLimit:LIMIT-1,
    paidWork:paidWorkEnabled===null?'unavailable':paidWorkEnabled?'enabled':'held',
    payoutCertification:project==='scaled-circle'?{onboarding:'founder_certified',readiness:'founder_certified',cashoutAndBankReceipt:'pending',
      evidenceKind:'Founder observation, September 19, 2026; not a provider readiness override'}:null,
    businesses:businesses.filter(Boolean),
    billing:{teamIssues:teamInventories?.every(Boolean)?teamInventories.filter(t=>t.needsReview).length:null,comped:count(subscriptions,s=>s.comped===true),plans:subscriptions?.map(s=>({businessId:s.docId,plan:label(s.planId||s.plan),status:label(s.status),comped:s.comped===true}))??null,active:count(subscriptions,s=>['active','trialing'].includes(s.status)),
      paymentIssues:count(subscriptions,s=>['past_due','unpaid','incomplete','incomplete_expired'].includes(s.status)),
      cancellationScheduled:count(subscriptions,s=>s.cancelAtPeriodEnd===true||s.cancel_at_period_end===true),
      workspaceCount:workspaces?.length??null,
      reconciliationIssues:count(operations,o=>['failed','attention','review_required'].includes(o.state||o.status)),
      moneySummary:'Customer payments, worker reserves and ScaledCircle revenue are separate. See each authoritative payment timeline; no gross-payment revenue total is inferred.'},
    email:{mailboxes:mailboxes?.map(m=>({businessId:m.docId,status:label(m.status),health:label(m.health)}))??null,
      branding:project==='scaled-circle'?'verified_checkpoint':'unavailable',gmailReview:project==='scaled-circle'?'submitted_under_review':'unavailable',newCustomerOnboarding:'verify_configuration',demoVideo:project==='scaled-circle'?'recorded_submitted':'unavailable',
      evidenceKind:'September 20, 2026 Google verification checkpoint; data access under review, CASA not complete; not a live provider probe'},
    notificationFailures:count(notifications,n=>['provider_configuration','uncertain'].includes(n.status)||(n.status==='retryable'&&n.attempts>=3)),
    release:project==='scaled-circle'?require('./admin_launch_candidate_evidence'):null};
}
module.exports={load,research,count};
